src/test/java/com/epam/tc/hw4/steps/ActionStep.java

```diff
@@ -0,0 +1,57 @@
+package com.epam.tc.hw4.steps;
+
+
+import io.qameta.allure.Step;
+import org.openqa.selenium.WebDriver;
+
+public class ActionStep extends BaseStep {
+    public ActionStep(WebDriver webDriver) {
+        super(webDriver);
+    }
+
+    @Step("Open test site by URL")
+    public void openWebSite() {
+        mainPage.openMainPage();
+    }
+
+    @Step("Perform login as {username}|{password}")
+    public void performLogin(String user, String password) {
+        mainPage.login(user, password);
+    }
+
+    @Step("Switch to the {frame}")
+    public void switchToIframe(String iframe) {
+        mainPage.switchToFrame();
+    }
+
+    @Step("Switch to original window back")
+    public void switchToParentWindow() {
+        mainPage.switchToHomePage();
+    }
+
+    @Step("Open through the header menu Service -> Different Elements Page")
+    public void openDifferentElementsPage() {
+        headerMenu.clickServiceMenu();
+        headerMenu.clickOnDifferentElementsServiceElement();
```

Comment:
It's better to work with your page fragments via page, as you have your header menu on a page.